test/Chainweb/Test/Pact/RemotePactTest.hs

```diff
@@ -165,28 +228,47 @@ checkRequestKeys filePrefix rks = do
 checkResponse :: FilePath -> RequestKeys -> PollResponses -> IO TestTree
 checkResponse filePrefix rks (PollResponses theMap) = do
     let fp = filePrefix ++ "-expected-resp.txt"
-
-    let mays = map (\x -> HM.lookup x theMap) (_rkRequestKeys rks)
+    let mays = map (`HM.lookup` theMap) (_rkRequestKeys rks)
     let values = _arResult <$> catMaybes mays
     let bsResponse = return $ toS $ foldMap A.encode values
 
     return $ goldenVsString (takeBaseName fp) (testPactFilesDir ++ fp) bsResponse
 
-testUrl :: Port -> ChainwebVersion -> String -> Int -> BaseUrl
-testUrl thePort v release chainNum = BaseUrl
+getCwBaseUrl :: Port -> BaseUrl
+getCwBaseUrl thePort = BaseUrl
     { baseUrlScheme = Https
     , baseUrlHost = "127.0.0.1"
     , baseUrlPort = fromIntegral thePort
-    , baseUrlPath = "chainweb/"
-                  ++ release ++ "/"
-                  ++ toS (chainwebVersionToText v) ++ "/"
-                  ++ "chain/"
-                  ++ show chainNum ++ "/"
-                  ++ "pact" }
+    , baseUrlPath = "" }
 
 escapedCmd :: BS.ByteString
 escapedCmd = [r|{"cmds":[{"hash":"d0613e7a16bf938f45b97aa831b0cc04da485140bec11cc8954e0509ea65d823472b1e683fa2950da1766cbe7fae9de8ed416e80b0ccbf12bfa6549eab89aeb6","sigs":[{"addr":"368820f80c324bbc7c2b0610688a7da43e39f91d118732671cd9c7500ff43cca","sig":"71cdedd5b1305881b1fd3d4ac2009cb247d0ebb55d1d122a7f92586828a1ed079e6afc9e8b3f75fa25fba84398eeea6cc3b92949a315420431584ba372605d07","scheme":"ED25519","pubKey":"368820f80c324bbc7c2b0610688a7da43e39f91d118732671cd9c7500ff43cca"}],"cmd":"{\"payload\":{\"exec\":{\"data\":null,\"code\":\"(+ 1 2)\"}},\"meta\":{\"gasLimit\":100,\"chainId\":\"0\",\"gasPrice\":1.0e-4,\"sender\":\"sender00\"},\"nonce\":\"2019-03-29 20:35:45.012384811 UTC\"}"}]}|]
 
+type PactClientApi
```

Comment:
There's types for the API already in `Chainweb.Pact.RestAPI`, you should be able to use `client` w/ those

Reply:
Ok